src/main/resources/application.conf

```diff
@@ -58,3 +58,33 @@ security: {
 sodium: {
   libraryPath: "/usr/lib/x86_64-linux-gnu/libsodium.so.18"
 }
+
+# Customization on the underlying actor system
+
+akka {
+
+  # custom host pool for akka-http client connections used for streaming request/responses
+  # tune the configuration based on load-handling capability of tezos nodes
+  # refer to host-connection-pooling section in
+  # https://doc.akka.io/docs/akka-http/current/configuration.html
+  # for available properties and their meaning
+  tezos-streaming-client {
```

Comment:
Are these configs defaults? How did you come at this specific configuration as the right way to go?

Reply:
we started with a parallel collection on a forkjoin pool of _parallelism level of 32_ to achieve high throughput, while default `max-connections` for the client is `4`. I tried to keep the number of connections high because the idea would be to reuse the same pool for all streaming http-client requests. I might be overshooting a bit here, possibly any number between 5-10 might work the same for now

The `max-open-requests` is `32` by default and this configuration is a precautionary control to avoid overflowing the server with simultaneous requests (considered bad-practice). On my local trials to download accounts for blocks at level 1000 I immediately hit the limit (something between 40-50 requests were sent together). So I kept it to a decent value above this limit, always considering that this pool would be reused for any similar operation (needing to concurrently send a high number of http requests to tezos, without flooding it) for any endpoint.